tests/cases/compiler/checkSuperCallBeforeThisAccessing9.ts

```diff
@@ -0,0 +1,21 @@
+﻿// @allowJs: true
+// @checkJs: true
+// @noEmit: true
+
+// @filename: noSuperInJSDocExtends.js
+class Based { }
+/** @extends {Based} */
+class Derived {
+    constructor() {
+        this;
+        this.x = 10;
+        var that = this;
+    }
+}
+
+/** @extends {Based} */
+class Derived2 {
+    constructor() {
+        super();

```

Comment:
shouldn't this be an error with this change, since checkConstructorDeclaration will not recognise Derived2 as having an extends clause?

Reply:
you're right. I fixed that check too